zebra/zebra_vty.c

```diff
@@ -1479,125 +1594,289 @@ DEFUN (ip_nht_default_route,
 	return CMD_SUCCESS;
 }
 
-static void show_nexthop_group_out(struct vty *vty, struct nhg_hash_entry *nhe)
+static void show_nexthop_group_out(struct vty *vty, struct nhg_hash_entry *nhe,
+				       json_object *json_nhe_hdr)
 {
 	struct nexthop *nexthop = NULL;
 	struct nhg_connected *rb_node_dep = NULL;
 	struct nexthop_group *backup_nhg;
 	char up_str[MONOTIME_STRLEN];
 	char time_left[MONOTIME_STRLEN];
+	json_object *json_dependants = NULL;
+	json_object *json_depends = NULL;
+	json_object *json_nexthop_array = NULL;
+	json_object *json_nexthops = NULL;
+	json_object *json = NULL;
+	json_object *json_backup_nexthop_array = NULL;
+	json_object *json_backup_nexthops = NULL;
+	char json_buf[100];
 
-	uptime2str(nhe->uptime, up_str, sizeof(up_str));
 
-	vty_out(vty, "ID: %u (%s)\n", nhe->id, zebra_route_string(nhe->type));
-	vty_out(vty, "     RefCnt: %u", nhe->refcnt);
-	if (thread_is_scheduled(nhe->timer))
-		vty_out(vty, " Time to Deletion: %s",
-			thread_timer_to_hhmmss(time_left, sizeof(time_left),
-					       nhe->timer));
-	vty_out(vty, "\n");
+	uptime2str(nhe->uptime, up_str, sizeof(up_str));
 
-	vty_out(vty, "     Uptime: %s\n", up_str);
-	vty_out(vty, "     VRF: %s\n", vrf_id_to_name(nhe->vrf_id));
+	if (json_nhe_hdr)
+		json = json_object_new_object();
 
+	if (json) {
+		snprintf(json_buf, sizeof(json_buf), "%u", nhe->id);
+		json_object_string_add(json, "type",
+				       zebra_route_string(nhe->type));
+		json_object_int_add(json, "refCount", nhe->refcnt);
+		if (thread_is_scheduled(nhe->timer)) 
+			json_object_string_add(json, "timeToDeletion",
+		       	                thread_timer_to_hhmmss(time_left,
+		       	                sizeof(time_left), nhe->timer));
+		json_object_string_add(json, "uptime", up_str);
+		json_object_string_add(json, "vrf", vrf_id_to_name(nhe->vrf_id));
+		
+	} else {
+		vty_out(vty, "ID: %u (%s)\n", nhe->id, zebra_route_string(nhe->type));
```

Comment:
Wrong indentation?

Reply:
I have not changed the indentation. Non json output looks like this:
```
primary# show nexthop-group rib 
ID: 2 (zebra)
     RefCnt: 2
     Uptime: 00:06:36
     VRF: default
     Valid, Installed
     Interface Index: 22
           is directly connected, veth2 (vrf default)
```

Reply:
![Screenshot-2022-12-13-22:35:37](https://user-images.githubusercontent.com/3352707/207438163-ec7f9562-530c-4940-8b79-a7e8d588c2ff.png)

Sorry, I was about code indentation, not the output :)

Reply:
Fixed it